### PR TITLE
Update CONTRIBUTING files

### LINKED ISSUE #59: Improvement for CONTRIBUTING.md
Actual version of `CONTRIBUTING.md` is verry short!

----
# Contribute

Thanks for your support! Please open a pull request on GitHub!

----

I based myself on this rule when starting to work on the French translation, you can see what kind of misunderstanding it caused! ;)

I propose (i could open a MR if you agree) something like (not sure for my english!):

----
# Contribute to the documentation

Thanks for your support! You're welcome!

* First, take a look at the [issues](https://github.com/jamulussoftware/jamuluswebsite/issues) to check if there already one on the same subject.  
If yes, you can join the discussion adding your comment.  

* If no, start by openning a [new issue](https://github.com/jamulussoftware/jamuluswebsite/issues/new/choose) with your proposal.

* Then, you can submit your merge request with your adds or changes following the rules explained [here](https://jamulus.io/wiki/Contribution) and below:
  * If not doing yet, fork this repository
  * create a new branch from the up to date `changes` branch
  * make your changes
  * push the new branch into your repository
  * submit the merge request linking to the opened issue from your branch to the `changes`branch.
  * wait for the review and the merge…

**DO NOT MERGE YOUR WORK** into your local `changes`branch, only keep it in sync by pooling from the remote Jamulus repository.
**DO NOT WORK ON** your local `changes` branch, always works from a new branch.  
If the `changes`branch is modified before submitting your request, keep the local `changes`branch in sync by pooling from it, merge the local `changes` branch into your working branch, resolve conflicts and then push your changes and create the merge request.

Please note that all the discussions are in English. If it is not you're native language and are not fluent with it, it can dissuade you. Don't be afraid, you can use [DeepL](https://www.deepl.com) wich make good translations as a help support.

### PR DESCRIPTION
**Pull request by ann0see**
As discussed in #59 I've now added the proposal from @gegeweb to a new contributing file.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (ann0see) [inline CONTRIBUTING.md]: Do we really want to link this page? I don't feel comfortable by doing so. In fact, we should link this file on the mentioned site not the other way round, I think

File: CONTRIBUTING.md
Comment (ann0see) [inline CONTRIBUTING.md]: This could go to the website

File: CONTRIBUTING.md
Comment (ann0see) [inline CONTRIBUTING.md:19]: I hope this is clear now. We might need to clarify our process.

**Review by ann0see — COMMENTED** [review]
A few comments

File: CONTRIBUTING.md
Comment (gegeweb) [reply CONTRIBUTING.md]: > we should link this file on the mentioned site not the other way round, I think

I think too.

File: CONTRIBUTING.md
Comment (gegeweb) [reply CONTRIBUTING.md]: Yes.

File: CONTRIBUTING.md
Comment (trebmuh) [inline CONTRIBUTING.md]: "follow **those** instructions"

File: CONTRIBUTING.md
Comment (trebmuh) [inline CONTRIBUTING.md]: (typo) **Decide**

File: CONTRIBUTING.md
Comment (trebmuh) [inline CONTRIBUTING.md]: "**p**ull **r**equest": consistency, it's always written "**P**ull **R**equest" in this page, we should keep it that way.

**Comment by gilgongo** [comment]
How do I actually see this file using Jeckyl? I can't really assess it in Github.

File: CONTRIBUTING.md
Comment (trebmuh) [inline CONTRIBUTING.md]: "Website" shouldn't be capitalised unless there is a good reason for that.
(Hint: it's not capitalised on another occurrence above)

**Comment by ann0see** [comment]
No, you can't. You need to access it via https://github.com/jamulussoftware/jamuluswebsite/blob/upd_contribute/CONTRIBUTING.md

File: README.md
Comment (trebmuh) [inline README.md]: (nitpicking) comma after "Please open an issue"

**Review by trebmuh — CHANGES_REQUESTED** [review]
A few small things

File: CONTRIBUTING.md
Comment (ann0see) [reply CONTRIBUTING.md]: Should be done

File: CONTRIBUTING.md
Comment (ann0see) [reply CONTRIBUTING.md]: Done.

**Comment by trebmuh** [comment]
I'm a little uncomfortable with Deepl's unique promotion as it is now

Suggestion:
> You can use a reasonably good online translator (e.g. [DeepL](https://www.deepl.com), [Reverso](https://www.reverso.net/text_translation.aspx), ...) which might be able to translate your language to English."

Thoughts?

File: CONTRIBUTING.md
Comment (gilgongo) [inline CONTRIBUTING.md]: "Don't be afraid" feels rather odd. I'd just say "If you don't speak English fluently you can use DeepL"

File: CONTRIBUTING.md
Comment (ann0see) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md
Comment (gilgongo) [inline CONTRIBUTING.md]: "check if anybody has raised"

File: CONTRIBUTING.md
Comment (gilgongo) [reply CONTRIBUTING.md]: What does "please subscribe to the issue" mean?

File: CONTRIBUTING.md
Comment (ann0see) [reply CONTRIBUTING.md]: done, but what about "the following"?

Also "allright" (L8) doesn't look right.

File: CONTRIBUTING.md
Comment (ann0see) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md
Comment (ann0see) [reply CONTRIBUTING.md]: https://github.com/jamulussoftware/jamuluswebsite/pull/157#issuecomment-740173966

We might have to rethink this

**Comment by ann0see** [comment]
> I'm a little uncomfortable with Deepl's unique promotion as it is now

Deepl is quite good. But we can of course link a few other pages.

File: CONTRIBUTING.md
Comment (ann0see) [reply CONTRIBUTING.md]: Subscribe to issue

It's a GitHub feature

File: CONTRIBUTING.md
Comment (ann0see) [reply CONTRIBUTING.md]: Should be fixed

**Comment by trebmuh** [comment]
> Deepl is quite good. But we can of course link a few other pages.

I know it is (I use it quite often myself). What makes me uncomfortable is to promote only one of them when there are alternatives, some of which are also very good and are improving (they are moving more and more to AI).

**Comment by ann0see** [comment]
Yes. You're right. I've integrated your proposal now.

**Comment by trebmuh** [comment]
Awesome, thanks.
LGTM, I'll "github approve" it now.

**Review by trebmuh — APPROVED** [review]

File: CONTRIBUTING.md
Comment (ann0see) [reply CONTRIBUTING.md]: Any thoughts on this? I've moved some files on the website but I'm not sure if that isn't too much. (also removed a few SourceForge references since I think to make the line between development and Forum clearer, most feature requests should go to github)

**Review by gilgongo — COMMENTED** [review]
You can make this live I guess, but it's in such a different written style and structure to the rest of the site - it really needs to be re-written in my opinion. And even then it makes the whole contribution process look really complicated if you ask me.

File: CONTRIBUTING.md
Comment (gilgongo) [reply CONTRIBUTING.md]: Yes, but why do we ask them to do this? It seems oddly specific.

**Comment by ann0see** [comment]
> You can make this live I guess

Are you sure about that? We will start the translation period as soon as gilgongo opens and merges the pr to translation. Every change until this will be included in the next version.

File: CONTRIBUTING.md
Comment (ann0see) [reply CONTRIBUTING.md]: @gegeweb?

I think it's since a user will then be able to receive updates on this issue. If you want to remove it, just do so.

**Comment by gilgongo** [comment]
Sorry I meant we can merge this. But really I'd prefer not to because I think it's on the wrong track. Confronting people with a big bunch of complex rules isn't going to encourage much contribution. Just tell them to post on the issues list and we can take it from there.

**Comment by ann0see** [comment]
Agree. Nevertheless we must at least make all the branches stuff clear. So this PR will **not** be merged for the next version.

File: CONTRIBUTING.md
Comment (ignotus666) [reply CONTRIBUTING.md]: "Follow **these** instructions" if you're going to list them below. "Follow the following instructions" is a bit repetitive.

The preferred form is "all right", two words.

File: CONTRIBUTING.md
Comment (gegeweb) [reply CONTRIBUTING.md]: > I think it's since a user will then be able to receive updates on this issue. If you want to remove it, just do so.

Sorry, I wasn't online last night.
I'll let you decide (it seems to be updated), it's not necessarily useful to keep it.

**Review by gegeweb — APPROVED** [review]
Seems good to me.

Thanks!

**Comment by ann0see** [comment]
@gilgongo @gegeweb 

I think we must re think the structure/wording etc. @gilgongo what would you remove?

**Comment by trebmuh** [comment]
Just to clarify things and to ensure that everybody is on the same page with this CONTRIBUTING.md file:
* we are talking here about a file which is not displayed on the jamulus.io website (this CONTRIBUTING.md file isn't the [Contribution website page](https://jamulus.io/wiki/Contribution))
* we should clarified who is the "targeted" audience here
* we should care to not overlap things between this CONTRIBUTING.md github repo page and the [Contribution website page](https://jamulus.io/wiki/Contribution)

I'd suggest that we:
* stay easy and simple on the [Contribution website page](https://jamulus.io/wiki/Contribution) which is intended for the basic users (since it's in the manual) and where we can write something along what @gilgongo proposed above:
> Just tell them to post on the issues list and we can take it from there.
* use the CONTRIBUTING.md file to write down our "internal" workflow, ie: how we are technically doing things, it'll help us to know (I mean "us" = the 4 or 5 current regular people here, so far: gilgongo ann0see ignotus666 gegeweb and yours truly)

I do agree with what you said above @gilgongo :
> Confronting people with a big bunch of complex rules isn't going to encourage much contribution

that said, the rules of how we do things here in this repo **are** complex and we need a document explaining that clearly and precisely. Without such a document, we're loosing time (with committing against the wrong branch as an example). As far as I understand it, this CONTRIBUTING.md page we're working on here isn't expected to be shown for the basic users, but is targeted at us.

Thoughts?

**Comment by ann0see** [comment]
> we are talking here about a file which is not displayed on the jamulus.io website (this CONTRIBUTING.md file isn't the Contribution website page)

Initially, yes. But it has changed for contribution in general. In my opinion two PRs or issues don't make sense since it's the same topic.

> we should clarified who is the "targeted" audience here

Yes. Absolutely. 

> stay easy and simple on the Contribution website page 

Yes. Nevertheless, we should add a (not necessarily prominent) link to users who want to actively contribute (in spirit of FOSS).

> Confronting people with a big bunch of complex rules isn't going to encourage much contribution

Agree.

>  the rules of how we do things here in this repo are complex

Yes. 

We must also clarify on a "who does what" policy.

I for example can only check de, en and to some degree fr Pull Requests (but would like to focus on de since it's my native toung). Some translators (the trreviewers group) should now have write access (at least to translate and changes). They should therefore merge these languages to translation. But that's another issue.

**Comment by gilgongo** [comment]
@trebmuh 

> we are talking here about a file which is not displayed on the jamulus.io website

OK I hadn't properly understood that. I agree we need to record what our process is, if only for the process of translation, which is not simple and we might forget the details. 

If it's basically a "private" page then it doesn't make sense to worry too much about style or architecture in that case.

**Comment by ann0see** [comment]
> we are talking here about a file which is not displayed on the jamulus.io website

This is not completely true since the branch and therefore this PR also includes changes to the contribution file on jamulus.io. Both files should be changed and we have to agree what to put where

**Comment by gegeweb** [comment]
> If it's basically a "private" page then it doesn't make sense to worry too much about style or architecture in that case.

I think the person who will read this will come from the "Contributing" page on the website, either directly, or after opening an issue.
So, having the skills to deal with Github, undestand "branch", "PR"… Processes should be more clear and detailed as possible here!

**Comment by gilgongo** [comment]
> I think the person who will read this will come from the "Contributing" page on the website

Very, very few people in the outside world have any clue about PRs, branches or anything else to do with git.

If we let such people discover a long detailed document about contribution that may well mean nothing to them, then that's bad because they'll probably think they won't be able to contribute when in fact they can. 

It would be better to let them introduce themselves and their ideas on the forums or Github issues, and we can then give them that link if we think it's appropriate, or simply implement their ideas on their behalf.

**Comment by trebmuh** [comment]
> > we are talking here about a file which is not displayed on the jamulus.io website
> 
> This is not completely true since the branch and therefore this PR also includes changes to the contribution file on jamulus.io. Both files should be changed and we have to agree what to put where

In my previous message, I've been clearly mentioning that I was talking about the CONTRIBUTING.md page to disambiguate it.

**Comment by ann0see** [comment]
Sorry about that @trebmuh I didn't recognise what gilgongo quoted so I thought he (and probably you) referred to the PR as a whole. Now it's clear to me too.

**Comment by trebmuh** [comment]
No worries at all. Github's thread system have some limitations.
Thanks for the message.

**Comment by ann0see** [comment]
I've added this and a few other features to the "Project" boards in GitHub to give this all a bit more structure.

Do we have any updates on this?

**Comment by ann0see** [comment]
I would maybe remove the "I don't know GitHub. What do I need to know?" which is shown on the website. Any thoughts on this?

File: CONTRIBUTING.md
Comment (ignotus666) [inline CONTRIBUTING.md]: Suggestion: 

'We will bundle them (if everything is all right) with any other changes and upload all of them as a combined Pull Request to jamulus.io.

File: wiki/en/en-Contribution.md
Comment (ignotus666) [inline wiki/en/en-Contribution.md]: 'We will then take care of your changes.'

File: CONTRIBUTING.md
Comment (ann0see) [reply CONTRIBUTING.md]: Changed.

File: wiki/en/en-Contribution.md
Comment (ann0see) [reply wiki/en/en-Contribution.md]: Done. We have to decide [what to do with this part](https://github.com/jamulussoftware/jamuluswebsite/pull/157#issuecomment-746959506). Do we really need to be that specific?

File: wiki/en/en-Contribution.md
Comment (ignotus666) [inline wiki/en/en-Contribution.md]: 'Decide which branch you have to commit your changes to' (One 'o').

'up-to-date' (hyphenated when preceding a noun).

File: CONTRIBUTING.md
Comment (gilgongo) [inline CONTRIBUTING.md]: I'd remove "Therefore have a look at the guideline below." as redundant. Assuming they're at all interested, nobody is going to suddenly stop reading.

File: CONTRIBUTING.md
Comment (gilgongo) [inline CONTRIBUTING.md]: "Is your content in a language that already exists on the site?" This reads a bit better.

File: CONTRIBUTING.md
Comment (gilgongo) [inline CONTRIBUTING.md]: This No: " "Yes: " style is a bit weird, particularly as we don't use it in the section above. I'd keep it to "If not..." and "If so..." really. It's also slightly ambiguous: "**No:** A fix which doesn't need to be translated" could be taken to mean "**Don't** make a fix that doesn't need to be translated". It's just odd!

File: README.md
Comment (gilgongo) [inline README.md]: "Contribute **a** translation"?

Wait - none of this stuff is here: https://github.com/jamulussoftware/jamuluswebsite/blob/upd_contribute/CONTRIBUTING.md

I'm confused by GitHub again... Giving up.

Comment (ann0see) [reply README.md]: Don't give up! These are in total 3 files. The file you saw here is the README file which links to the contributing file.

File: wiki/en/en-Contribution.md
Comment (ann0see) [reply wiki/en/en-Contribution.md]: Done.

File: CONTRIBUTING.md
Comment (ann0see) [reply CONTRIBUTING.md]: Yes. Removed this

File: README.md
Comment (ann0see) [reply README.md]: Yes,
> Contribute a translation

is better since we talk about translation afterwards.

**Comment by ann0see** [comment]
I think, we should merge this and then do some changes afterwards. What do you think?

**Comment by ann0see** [comment]
I might not be available in the next few days. @gilgongo and others  free to merge/modify my PRs (especially this one). 

The next steps would then be to look through the wiki to fix syntax errors and typos. @ignotus666 wanted to do that.

**Comment by ignotus666** [comment]
I'll see if I can do it this coming week.

**Comment by ann0see** [comment]
I will merge this in a few minutes and then we can discuss more of it

**Comment by ann0see** [comment]
@gilgongo now reviewing it should be easier since it's on the changes branch

**Pull request merged**
